default_settings.txt
-- Global configuration variables

-- set this to true to make moretrees spawn saplings at mapgen time instead
-- of fully-grown trees, which will grow into full trees after a very short
-- delay.  This reduces mapgen lag in some situations, and fixes situations
-- where the mapgen conflicts with the tree generator.

moretrees.spawn_saplings		= true

-- Set this to true if you want the plantlike drawtype for leaves, which
-- improves some peoples' framerates without resorting to making leaf nodes opaque.
-- Affects default leaves and default jungle leaves also.

moretrees.plantlike_leaves		= false

-- Enable this if you want moretrees to redefine default apples so that they
-- fall when leaves decay/are dug.

moretrees.enable_redefine_apple		= true

-- Set this to true to enable leaf decay of all trees except the default ones.

moretrees.enable_leafdecay		= true

-- various related settings to configure leaf decay.

moretrees.leafdecay_delay		= 2
moretrees.leafdecay_chance		= 5
moretrees.leafdecay_radius		= 5
moretrees.palm_leafdecay_radius		= 10
moretrees.date_palm_leafdecay_radius	= 14

-- Change these settings if you want default trees to be gradually cut down
-- above the elevation where firs normally generate.

moretrees.firs_remove_default_trees	= false
moretrees.firs_remove_interval		= 2
moretrees.firs_remove_chance		= 150

-- Cocos palm settings

moretrees.coconuts_regrow		= true
moretrees.coconuts_convert_existing_palms = true	-- Converting existing palm trees will make coconuts regrow on them as well
							-- Else, they will only regrow on newly-spawned palms
							-- However, conversion is not an exact science, and although an attempt is
							-- made to detect whether a trunk belongs to an actual palm, some coconut trunks
							-- and some coconuts may be incorrectly converted.
moretrees.coconut_flower_interval	= 59
moretrees.coconut_flower_chance		= 67
moretrees.coconut_grow_interval		= 2 * moretrees.coconut_flower_interval * moretrees.coconut_flower_chance
							-- Actual interval will randomly vary between 67% and 133% of this value
							-- 2 * 59 * 67 ~ 2 hours. So flowers become coconuts in about 6 hours
moretrees.coconut_item_drop_ichance	= 10 		-- inverse probability of ripe coconuts dropping as items (instead of disappearing)

-- Date palm settings

-- Suggested configuration alternatives:
-- - Dates grow only when pollinated:
--   - Set dates_regrow_pollinated to true
--   - Set dates_regrow_unpollinated_percent to 0
-- - Dates grow without pollination. Pollination disabled:
--   - Set dates_regrow_pollinated to false
--   - Set dates_regrow_unpollinated_percent to some larger positive value, e.g. 95
-- - Dates grow, but more and faster if male flowers are nearby
--   - Set dates_regrow_pollinated to true
--   - Set dates_regrow_unpollinated_percent to some small positive value, e.g. 33
--   - Optional but recommended: Reduce the pollination distance, e.g. to 30

-- Note that it should not be necessary to disable pollination for performance
-- reasons. A lot of effort has gone into ensuring that date growing will not cause lag.
--
-- If lag is suspected, use the chat command '/dates_stats' to obtain the male dates
-- search time, as well as the counts of total number of searches requested and the
-- number of searches actually performed.

moretrees.dates_regrow_pollinated	= true		-- Enable pollination. If enabled, male trees are required for dates to grow.
							-- If disabled, dates_regrow_unpollinated_percent must be non-zero for dates to regrow.
moretrees.dates_regrow_unpollinated_percent	= 0	-- Percentage of female dates becoming dates without being pollinated.
							-- If 0, dates_regrow_pollinated must be enabled for dates to grow.
moretrees.dates_female_percent		= 57		-- Ratio of female to male trees - tune this to improve # of generated trees that actually bear fruit
							-- ~57% gives near optimal results for groups of 3 random trees, while it is only slightly suboptimal
							-- for groups of 2 and 4 random trees (~2% less fruit than optimal).
							-- Optimal values per group size: 2: 50%, 3: 57.78%, 4: 63%, 5: 66.9%, 6: 69.9%, [...], 12: 79.8%
							-- So 57% is optimal for small groups of trees. As larger groups have more female palms anyway, a
							-- less than optimal proportion of female to male trees is not a problem.
moretrees.dates_pollination_distance	= 120
moretrees.dates_blossom_search_time_treshold = 1000	-- If average male blossom search time (in microseconds) exceeds this, start limiting the search load.
moretrees.dates_blossom_search_iload = 10		-- Inverse fraction of CPU time that male blossom searching search may consume.
							-- As searching a large area (radius: dates_pollination_distance/3 per attempt) can cause lag,
							-- this limits the search frequency server-wide so that the impact on server lag is minimised
							-- For personal servers, this can be set lower, or even to 1 or 0 (0 disables load limiting).
							-- Obtain the current average search time using /dates_stats
moretrees.dates_flower_interval		= 59
moretrees.dates_flower_chance		= 181
moretrees.dates_grow_interval		= 2 * moretrees.dates_flower_interval * moretrees.dates_flower_chance
							-- As date palms have a high yield, don't grow dates too fast
							-- The actual interval will vary randomly between 67% and 133% of this value.
							-- 2 * 59 * 181 ~ 6 hours. So by default flowers become dates in about one (human) day.
moretrees.dates_item_drop_ichance	= 10 		-- inverse probability of ripe dates dropping as items (instead of disappearing)

-- Sapling settings

moretrees.sapling_interval		= 100
moretrees.sapling_chance		= 5

-- Enable this only if you have used an old moretrees version which was using biome_lib
-- and when you notice large areas with ongen saplings that don't grow
moretrees.grow_legacy_saplings	= false

-- If this variable is set to true, drop leaves out as entities during leaf
-- decay, rather than just disappearing them.

moretrees.decay_leaves_as_items = false


settings.lua
local stg = minetest.settings

--spawns tree on mapgen as saplings, legacy setting
moretrees.spawn_saplings		= stg:get_bool("moretrees.spawn_saplings", true)

--enable plantlike drawtype for leaves
moretrees.plantlike_leaves		= stg:get_bool("moretrees.plantlike_leaves", false)

-- Enable this if you want moretrees to redefine default apples so that they
-- fall when leaves decay/are dug.
moretrees.enable_redefine_apple		= stg:get_bool("moretrees.enable_redefine_apple", true)

-- various related settings to configure leaf decay.
moretrees.leafdecay_radius		= tonumber(stg:get("moretrees.leafdecay_radius")) or 5
moretrees.palm_leafdecay_radius		= tonumber(stg:get("moretrees.palm_leafdecay_radius")) or 10
moretrees.date_palm_leafdecay_radius	= tonumber(stg:get("moretrees.date_palm_leafdecay_radius")) or 14

-- Cocos palm settings
moretrees.coconuts_regrow		= stg:get_bool("moretrees.coconuts_regrow", true)
moretrees.coconuts_convert_existing_palms = stg:get_bool("moretrees.coconuts_convert_existing_palms", true)
moretrees.coconut_flower_interval	= tonumber(stg:get("moretrees.coconut_flower_interval")) or 59
moretrees.coconut_flower_chance		= tonumber(stg:get("moretrees.coconut_flower_chance")) or 67
moretrees.coconut_grow_interval		= tonumber(stg:get("moretrees.coconut_grow_interval")) or
    2 * moretrees.coconut_flower_interval * moretrees.coconut_flower_chance
moretrees.coconut_item_drop_ichance	= tonumber(stg:get("moretrees.coconut_item_drop_ichance")) or 10

-- Date palm settings
moretrees.dates_regrow_pollinated	= stg:get_bool("moretrees.dates_regrow_pollinated", true)
moretrees.dates_regrow_unpollinated_percent	= tonumber(stg:get("moretrees.dates_regrow_unpollinated_percent")) or 0
moretrees.dates_female_percent		= tonumber(stg:get("moretrees.dates_female_percent")) or 57
moretrees.dates_pollination_distance	= tonumber(stg:get("moretrees.dates_pollination_distance")) or 120
moretrees.dates_blossom_search_time_treshold = tonumber(stg:get("moretrees.dates_blossom_search_time_treshold")) or 1000
moretrees.dates_blossom_search_iload = tonumber(stg:get("moretrees.dates_blossom_search_iload")) or 10
moretrees.dates_flower_interval		= tonumber(stg:get("moretrees.dates_flower_interval")) or 59
moretrees.dates_flower_chance		= tonumber(stg:get("moretrees.dates_flower_chance")) or 181
moretrees.dates_grow_interval		= tonumber(stg:get("moretrees.dates_grow_interval")) or
    2 * moretrees.dates_flower_interval * moretrees.dates_flower_chance
moretrees.dates_item_drop_ichance	= tonumber(stg:get("moretrees.dates_item_drop_ichance")) or 10

-- Sapling settings
moretrees.sapling_interval		= 100
moretrees.sapling_chance		= 5

-- Enable this only if you have used an old moretrees version which was using biome_lib
-- and when you notice large areas with ongen saplings that don't grow
moretrees.grow_legacy_saplings	= stg:get_bool("moretrees.grow_legacy_saplings", false)

-- If this variable is set to true, drop leaves out as entities during leaf
-- decay, rather than just disappearing them.
moretrees.decay_leaves_as_items = stg:get_bool("moretrees.decay_leaves_as_items", false)

-- this is an internal setting for games that only have one type of planks
moretrees.enable_planks = xcompat.gameid ~= "voxelgarden"